tests/NuGet.IndexingTests/LatestListedHandlerTests.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Lucene.Net.Documents;
+using Lucene.Net.Search;
+using NuGet.Indexing;
+using NuGet.IndexingTests.TestSupport;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class LatestListedHandlerTests
+    {
+        [Theory]
+        [MemberData(nameof(DocumentsAreFilteredData))]
+        public void DocumentsAreFilteredTest(
+            bool includeUnlisted,
+            bool includePrerelease,
+            bool includeSemVer2,
+            string readerName,
+            string id,
+            Document[] documents,
+            NuGetVersion[] versions,
+            int numSubReadersToFake,
+            int expectedNumDocsInResult)
+        {
+            var handler = new LatestListedHandler(includeUnlisted, includePrerelease, includeSemVer2);
```

Comment:
Tests (especially longer ones) should have Arrange, Act, and Assert sections (denoted by comments).